### PR TITLE
fix: rename INSTALLED_COMPONENT_NOT_FOUND as COMPONENT_LOAD_FAILURE; …

### PR DESCRIPTION
**Pull request by indougnito**
…create separate error code CUSTOM_PLUGIN_NOT_SUPPORTED

**Issue #, if available:**

**Description of changes:**

Rename INSTALLED_COMPONENT_NOT_FOUND as COMPONENT_LOAD_FAILURE.

For scenarios where the user attempts to install a custom plugin, created a new error code CUSTOM_PLUGIN_NOT_SUPPORTED with type USER_COMPONENT_ERROR.

**Why is this change necessary:**

INSTALLED_COMPONENT_NOT_FOUND is a misleading code for ServiceLoadException, which is thrown when the component is found but there is a loading issue.

Attempting to install a custom plugin should be a USER_COMPONENT_ERROR, not a NUCLEUS_ERROR; hence the need for a new error code CUSTOM_PLUGIN_NOT_SUPPORTED, distinct from COMPONENT_LOAD_FAILURE.

**How was this change tested:**
- [ ] Updated or added new unit tests.
- [ ] Updated or added new integration tests.
- [ ] Updated or added new end-to-end tests.
- [ ] If my code makes a remote network call, it was tested with a proxy.

**Any additional information or context required to review the change:**

**Documentation Checklist:**
 - [ ] Updated the README if applicable.

**Compatibility Checklist:**
- [ ] I confirm that the change is backwards compatible.
- [ ] Any modification or deletion of public interfaces does not impact other plugin components.
- [ ] For external library version updates, I have reviewed its change logs and Nucleus does not consume 
  any deprecated method or type.

Refer to [Compatibility Guidelines](/COMPATIBILITY.md) for more information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by MikeDombo** [comment]
We identified new issues on unchanged lines of code. Navigate to the Amazon CodeGuru Reviewer console to view the recommendations to fix them.

File: src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java [hunk 1]
```diff
@@ -536,7 +537,7 @@ private Class<?> locateExternalPlugin(String name, Topics serviceRootTopics) thr
         if (storedDigest == null || storedDigest.getOnce() == null) {
             logger.atError("plugin-load-error").kv(GreengrassService.SERVICE_NAME_KEY, name)
                     .log("Local external plugin is not supported by this greengrass version");
-            throw new ServiceLoadException("Custom plugins is not supported by this greengrass version");
+            throw new CustomPluginNotSupportedException("Custom plugins is not supported by this greengrass version");
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java:540]: let's fix this error message. `Locally deployed plugin components are not supported. Plugins must be deployed via a cloud-based deployment.`

Comment (indougnito) [reply src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java:540]: updating

**Review by MikeDombo — APPROVED** [review]

**Comment by indougnito** [comment]
force-squashed to appease commit message linter

**Comment by indougnito** [comment]
shortened commit message to appease linter

**Comment by indougnito** [comment]
fix PluginComponentTest.GIVEN_kernel_WHEN_locate_plugin_without_digest_THEN_plugin_is_not_loaded_into_JVM

**Comment by indougnito** [comment]
fixed on windows: UnloadableServiceIntegTest.GIVEN_unloadable_plugin_digest_missing_WHEN_nucleus_launch_THEN_nucleus_starts_and_other_services_running

**Review by junfuchen99 — APPROVED** [review]

**Review by alter-mage — APPROVED** [review]

**Pull request merged**